### PR TITLE
Fix #681 - "Variable not found" error in ggmarginal

### LINKED ISSUE #681: "Variable not found" error in ggmarginal
Code:

```python
ggplot({'x': [], 'y': []}, aes('x', 'y')) + \
    geom_point() + \
    ggmarginal("t", layer=geom_histogram())
```

Result:

`Variable not found: '..density..'. Variables in data frame: ['x', 'y']`

But:

- the library should work with empty data and
- shouldn't complain about `'..density..'` when current stat is `'..count..'`

### PR DESCRIPTION
**Pull request by IKupriyanov-HORIS**

### CODE REVIEW

File: plot-config-portable/src/jvmTest/kotlin/plot/config/EdgeCasesTest.kt [hunk 1]
```diff
@@ -133,6 +133,62 @@ class EdgeCasesTest {
         (GeomName.values() - listOf(LIVE_MAP, IMAGE)).forEach(::checkGeom)
     }
 
+    @Test
```
Comment (alshan) [inline plot-config-portable/src/jvmTest/kotlin/plot/config/EdgeCasesTest.kt:136]: What does this have to do with "Fix #681 - "Variable not found" error in ggmarginal"?

File: plot-config-portable/src/jvmTest/kotlin/plot/config/ScaleOrderingTest.kt [hunk 2]
```diff
@@ -15,6 +15,7 @@ import jetbrains.datalore.plot.common.color.ColorPalette
 import jetbrains.datalore.plot.config.TestUtil.getSingleGeomLayer
 import kotlin.test.Test
 import kotlin.test.assertEquals
+import kotlin.test.assertNull
 
 
 class ScaleOrderingTest {
```
Comment (alshan) [inline plot-config-portable/src/jvmTest/kotlin/plot/config/ScaleOrderingTest.kt:21]: What does scale ordering tests have to do with "Fix #681 - "Variable not found" error in ggmarginal"?

**Review by alshan — CHANGES_REQUESTED** [review]

File: plot-config-portable/src/jvmTest/kotlin/plot/config/EdgeCasesTest.kt (hunk 1 above)
Comment (IKupriyanov-HORIS) [reply plot-config-portable/src/jvmTest/kotlin/plot/config/EdgeCasesTest.kt:136]: ggmarginal is not the root cause of the #681. The real issue is this - when stat is failed to apply it returns empty stats with vars that only have default mappings, missing other stat vars. 

So #681 is actually about `layer=geom_histogram()`, implicit using of the `..density..` var (that doesn't have default mapping) and empty data.
 
This test shows another, but more complex case - when data is not empty, but smooth stat still failed to apply (data points count is less than 3) and yield empty stats without the `..se..` var.

File: plot-config-portable/src/jvmTest/kotlin/plot/config/ScaleOrderingTest.kt (hunk 2 above)
Comment (IKupriyanov-HORIS) [reply plot-config-portable/src/jvmTest/kotlin/plot/config/ScaleOrderingTest.kt:21]: Test expected that plot with empty data will provide scale with empty breaks. This PR changes this behaviour and now plot doesn't provide scale for empty data. So this test started to fail and it was fixed.

**Pull request merged**